src/modules/transactions/services/methods.ts

```diff
@@ -128,13 +128,11 @@ export class TransactionService {
       },
     });
 
-    const { gasPrice, usedFee, minFee } =
+    const { maxFee } =
       await vault.provider.getTransactionCost(transactionRequest);
```

Comment:
In local tests, the `maxFee` is the exact price of fee

Reply:
use maxGasPerPredicate